src/components/carousel/carousel.ts

```diff
@@ -0,0 +1,385 @@
+
+  private _observerCallback({
+    changes: { added, attributes },
+  }: MutationControllerParams<IgcCarouselSlideComponent>) {
+    const activeSlides = this.slides.filter((slide) => slide.active);
+
+    if (activeSlides.length <= 1) {
+      return;
+    }
+
+    const carouselSlides = this.slides;
+    const idx = carouselSlides.indexOf(
+      added.length ? added.at(-1)! : attributes.at(-1)!
+    );
+
+    for (const [i, slide] of carouselSlides.entries()) {
+      if (slide.active && i !== idx) {
+        slide.active = false;
+      }
+    }
+  }
+
+  /**
+   * Whether the carousel should skip rotating to the first slide after it reaches the last.
+   * @attr skip-loop
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'skip-loop' })
+  public skipLoop = false;
+
+  /**
+   * Whether the carousel should ignore use interactions and not pause on them.
+   * @attr skip-pause-on-interaction
+   */
+  @property({
+    type: Boolean,
+    reflect: true,
+    attribute: 'skip-pause-on-interaction',
+  })
+  public skipPauseOnInteraction = false;
+
+  /**
+   * Whether the carousel should skip rendering of the default navigation buttons.
+   * @attr skip-navigation
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'skip-navigation' })
+  public skipNavigation = false;
+
+  /**
+   * Whether the carousel should render the picker controls (dots).
+   * @attr with-picker
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'with-picker' })
+  public withPicker = false;
+
+  /**
+   * The carousel alignment.
+   * @attr vertical
+   */
+  @property({ type: Boolean, reflect: true })
+  public vertical = false;
+
+  /**
+   * Sets the orientation of the picker controls (dots).
+   * @attr
+   */
+  @property({ reflect: true, attribute: 'indicators-orientation' })
+  public indicatorsOrientation: 'start' | 'end' = 'end';
+
+  /**
+   * The duration in milliseconds between changing the active slide.
+   * @attr interval
+   */
+  @property({ type: Number, reflect: false })
+  public interval!: number;
+
+  /**
+   * The total number of slides.
+   */
+  @property({ type: Number, reflect: false, attribute: false })
+  public get total(): number {
+    return this.slides.length;
+  }
+
+  /**
+   * The index of the current active slide.
+   */
+  @property({ type: Number, reflect: false, attribute: false })
+  public get current(): number {
+    return this.slides.findIndex((slide) => slide.active) ?? 0;
+  }
+
+  /**
+   * The slides of the carousel.
+   */
+  @queryAssignedElements({ selector: IgcCarouselSlideComponent.tagName })
+  public slides!: Array<IgcCarouselSlideComponent>;
+
+  /**
+   * Whether the carousel is in playing state.
+   */
+  @property({ type: Boolean, reflect: false, attribute: false })
+  public isPlaying = false;
+
+  /**
+   * Whether the carousel in in paused state.
+   */
+  @property({ type: Boolean, reflect: false, attribute: false })
+  public isPaused = false;
```

Comment:
These should be getters that reflect the internal state of the carousel and are not reactive properties.